### PR TITLE
Move text references to index created version to IndexVersion

### PR DESCRIPTION
**Pull request by thecoop**
Migrate text references to `index.created.version` to use `IndexVersion`

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/example-plugins

File: plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ public void testAuthorizeIndexAction() {
         CustomAuthorizationEngine engine = new CustomAuthorizationEngine();
         Map<String, IndexAbstraction> indicesMap = new HashMap<>();
         indicesMap.put("index", new ConcreteIndex(IndexMetadata.builder("index")
-            .settings(Settings.builder().put("index.version.created", Version.CURRENT))
+            .settings(Settings.builder().put("index.version.created", IndexVersion.current()))
```
Comment (ldematte) [inline plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java:121]: Can we change this to `IndexMetadata.SETTING_VERSION_CREATED` too?

File: server/src/test/java/org/elasticsearch/index/IndexSettingsTests.java [hunk 2]
```diff
@@ -729,7 +729,7 @@ public void testArchiveBrokenIndexSettings() {
 
         settings = IndexScopedSettings.DEFAULT_SCOPED_SETTINGS.archiveUnknownOrInvalidSettings(
             Settings.builder()
-                .put("index.version.created", Version.CURRENT.id) // private setting
+                .put(IndexMetadata.SETTING_VERSION_CREATED, IndexVersion.current()) // private setting
```
Comment (ldematte) [inline server/src/test/java/org/elasticsearch/index/IndexSettingsTests.java:732]: Is this missing a `.toString()` or is it not important?

File: plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java (hunk 1 above)
Comment (thecoop) [reply plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java:121]: This is in a plugin example, not compiled code, so I don't want to introduce class references that might not be there when this is used

**Comment by thecoop** [comment]
@elasticmachine update branch

**Review by ldematte — APPROVED** [review]
LGTM

**Pull request merged**